### PR TITLE
Add Laplacian Score

### PR DESCRIPTION
**Pull request by annbumagina**

### CODE REVIEW

File: ITMO_FS/filters/measures.py [hunk 1]
```diff
@@ -385,6 +387,43 @@ def pearson_corr(X, y):
     return (sum_dev / np.sqrt(np.sum(sq_dev_y) * np.sum(sq_dev_x))).reshape((-1,))
 
 
+def euclidean_distance(point1, point2):
```
Comment (LastShekel) [inline ITMO_FS/filters/measures.py]: suggest you to use  numpy.linalg.norm 
it is optimized and works well

File: ITMO_FS/filters/measures.py [hunk 2]
```diff
@@ -385,6 +387,43 @@ def pearson_corr(X, y):
...
+    for i in range(len(point1)):
+        sum_squared_distance += (point1[i] - point2[i]) * (point1[i] - point2[i])
+    return sqrt(sum_squared_distance)
+
+
+def laplacian_score(X, y, k_neighbors = 5, t = 1, 
+                    metric = euclidean_distance, **kwargs):
+    n, m = X.shape
+    k_neighbors = min(k_neighbors, n - 1)
+    if 'model' in kwargs.keys():
```
Comment (LastShekel) [inline ITMO_FS/filters/measures.py]: what is considered as model?

File: ITMO_FS/filters/measures.py [hunk 3]
```diff
@@ -385,6 +387,43 @@ def pearson_corr(X, y):
...
 
 
+def euclidean_distance(point1, point2):
+    sum_squared_distance = 0
+    for i in range(len(point1)):
+        sum_squared_distance += (point1[i] - point2[i]) * (point1[i] - point2[i])
+    return sqrt(sum_squared_distance)
+
+
+def laplacian_score(X, y, k_neighbors = 5, t = 1,
```
Comment (LastShekel) [inline ITMO_FS/filters/measures.py]: Add coments please 
take a look to MOSS.py or Add_del wrapper for examples

File: ITMO_FS/filters/measures.py (hunk 2 above)
Comment (annbumagina) [reply ITMO_FS/filters/measures.py]: weight matrix S of the graph that models the local structure of the data space
(p.2 in the paper)

File: ITMO_FS/filters/measures.py [hunk 4]
```diff
@@ -385,6 +386,67 @@ def pearson_corr(X, y):
...
+
+    Returns
+    -------
+    List of scores of each feature.
+    The smaller the laplacian score is, the more important the feature is.
+
+    See Also
+    --------
+    https://papers.nips.cc/paper/2909-laplacian-score-for-feature-selection.pdf
+
```
Comment (LastShekel) [inline ITMO_FS/filters/measures.py]: add some example

File: ITMO_FS/filters/measures.py [hunk 5]
```diff
@@ -385,6 +386,77 @@ def pearson_corr(X, y):
...
+                    S[i, j] = exp(-d * d / t)
+            distances.sort()
+            for j in range(k_neighbors):
+                S[i, distances[j][1]] = S[distances[j][1], i] = exp(-distances[j][0] * distances[j][0] / t)
+    ONE = np.ones((n,))
+    D = np.diag(S.dot(ONE))
+    L = D - S
+    LS = []
+    for r in range(m):
+        f = X[:, r] - ONE * (X[:, r].dot(D.dot(ONE)) / ONE.dot(D.dot(ONE)))
```
Comment (LastShekel) [inline ITMO_FS/filters/measures.py]: this loop can be redone with matrix functions using transposing

**Pull request merged**